app/components/app_search_component.rb

```diff
@@ -98,15 +98,15 @@ class AppSearchComponent < ViewComponent::Base
           <% if show_buttons_in_details? %>
             <div class="app-button-group">
               <%= f.govuk_submit "Update results", secondary: true, class: "app-button--small" %>
-              <%= govuk_button_link_to "Clear filters", @url, class: "app-button--small app-button--secondary" %>
+              <%= govuk_button_link_to "Clear filters", clear_filters_path, class: "app-button--small app-button--secondary" %>
             </div>
           <% end %>
         <% end %>
 
         <% unless show_buttons_in_details? %>
           <div class="app-button-group">
             <%= f.govuk_submit "Update results", secondary: true, class: "app-button--small" %>
-            <%= govuk_button_link_to "Clear filters", @url, class: "app-button--small app-button--secondary" %>
+            <%= govuk_button_link_to "Clear filters", clear_filters_path, class: "app-button--small app-button--secondary" %>
```

Comment:
Rather than specifying a path, I think it's possible to set the name of the button to `clear_filters` with a value of `true, I think that's a cleaner approach:

```suggestion
            <%= govuk_button_link_to "Clear filters", clear_filters_path, class: "app-button--small app-button--secondary", name: "clear_filters", value: "true" %>
```

Reply:
Unfortunately it doesn't work. It will only add html attributes to the link tag: Example:

`<a class="nhsuk-button app-button--small app-button--secondary" data-module="nhsuk-button" name="clear_filters" value="true" href="/sessions/session-2/consent" data-govuk-button-init="">Clear filters</a>`

Reply:
Oh yes, sorry, I missed that it was a link and thought it was a `<button>`